frontend_search_test.py

```diff
@@ -1,27 +1,36 @@
 from flask import Flask, render_template, flash, request
-from wtforms import Form, TextField, TextAreaField, validators, StringField, SubmitField
-from search import prepare_tree, search, find_artic,get_measure
+from wtforms import Form, TextField, validators
 from lxml import etree
+from search import prepare_tree, search, find_artic, get_measure
 
 """Turn user submitted MEI code snippet into xml file."""
+
+
 def transform_xml(xml_code):
     xml = xml_code.encode()
-    f = open('upload.xml', 'wb')
-    f.write(xml)
-    f.close()
+    file = open('upload.xml', 'wb')
```

Comment:
use context manager syntax instead:

```python
with open('upload.xml', 'wb') as file_descriptor:
    file_descriptor.write(xml)
```

Reply:
Also, is the 'b' in 'wb' necessary? xml should be mostly text, so it's unclear you need to write binary.